implementations/micrometer-registry-otlp/src/main/java/io/micrometer/registry/otlp/OtlpMetricsSender.java

```diff
@@ -15,25 +15,125 @@
 import java.util.Map;
+import java.util.Objects;
 
 /**
  * This is responsible for sending OTLP format metrics to a compatible location. Specific
  * implementations can use different transports or clients for sending the metrics.
  *
  * @since 1.15.0
  */
-public interface OtlpMetricsSender extends MetricsSender {
+public interface OtlpMetricsSender {
 
     /**
      * Send a batch of OTLP Protobuf format metrics to an OTLP receiver.
-     * @param address address of the OTLP receiver to which metrics will be sent
-     * @param metricsData OTLP protobuf encoded batch of metrics
-     * @param headers metadata to send as headers with the metrics data
-     * @throws Throwable when there is an exception in sending the metrics; the caller
+     * @param request metrics request to publish
+     * @throws Exception when there is an exception in sending the metrics; the caller
      * should handle this in some way such as logging the exception
      */
-    @Override
-    void send(String address, byte[] metricsData, Map<String, String> headers) throws Throwable;
+    void send(Request request) throws Exception;
+
+    /**
+     * Start building a request to use with this sender.
+     * @param metricsData metrics data that will be sent
+     * @return builder
+     * @see Request.Builder#send
+     */
+    default Request.Builder newRequest(byte[] metricsData) {
+        return new Request.Builder(metricsData, this);
+    }
+
+    /**
+     * Immutable representation of a payload of metrics to use with a
+     * {@link OtlpMetricsSender}.
+     *
+     * @since 1.15.0
+     */
+    class Request {
+
+        @Nullable
+        private final String address;
+
+        private final Map<String, String> headers;
+
+        private final byte[] metricsData;
+
+        /**
+         * Represents a payload of metrics to be sent.
+         * @param address where to send the metrics
+         * @param metricsData encoded batch of metrics
+         * @param headers metadata to send as headers with the metrics data
+         */
+        private Request(@Nullable String address, Map<String, String> headers, byte[] metricsData) {
+            this.address = address;
+            this.headers = headers;
+            this.metricsData = metricsData;
+        }
+
+        @Nullable
+        public String getAddress() {
+            return address;
+        }
+
+        public byte[] getMetricsData() {
+            return metricsData;
+        }
+
+        public Map<String, String> getHeaders() {
+            return headers;
+        }
+
+        @Override
+        public String toString() {
+            return "OtlpMetricsSender.Request for address: " + address + ", headers: " + headers + ", metricsData: "
+                    + new String(metricsData, StandardCharsets.UTF_8);
+        }
+
+        public static class Builder {
+
+            private final byte[] metricsData;
+
+            private final OtlpMetricsSender sender;
```

Comment:
I think removing this (the responsibility of the builder to send the request) as well as `newRequest()` from the sender would simplify things (the builder only builds and the sender only sends).

So the registry would do this:
```java
metricsSender.send(Request.builder(bytes).address(url).headers(headers).build());
```
instead of
```java
metricsSender.newRequest(request.toByteArray()).address(config.url()).headers(config.headers()).send();
```

Reply:
Sure, we can do that. I was trying to align somewhat with what is there in HttpSender and the fluent API has its advantages, but it's not worth distracting ourselves with this API so much when its usage will be largely internal. I'll update as you've suggested.